mhs/spineroutelookup/lookup/sds_connection_factory.py

```diff
@@ -41,7 +42,56 @@ def build_sds_connection_tls(ldap_address: str, private_key: str, local_cert: st
 
     ldap3.set_config_parameter('RESTARTABLE_TRIES', _LDAP_CONNECTION_RETRIES)
     server = ldap3.Server(ldap_address, use_ssl=True, tls=load_tls, connect_timeout=_LDAP_CONNECTION_TIMEOUT_IN_SECONDS)
-    logger.info('Opening LDAP connection using TLS')
-    connection = ldap3.Connection(server, auto_bind=True, client_strategy=ldap3.REUSABLE)
-    logger.info('LDAP connection successful')
+    logger.info('Configuring LDAP connection using TLS')
+
+    return _configure_ldap_connection(server)
+
+
+def _configure_ldap_connection(server) -> ldap3.Connection:
+    lazy_ldap = str2bool(config.get_config("LAZY_LDAP", default=str(True)))
+    if lazy_ldap:
+        connection = ldap3.Connection(server,
+                                      lazy=True,
```

Comment:
do we need lazy_ldap var?
cant we use:
lazy=config.get_config("LAZY_LDAP", default=str(True)) ?

Reply:
I can inline it into:
````python
def _configure_ldap_connection(server) -> ldap3.Connection:
    if str2bool(config.get_config("LAZY_LDAP", default=str(True))):
````
That would be better?